_pytest/fixtures.py

```diff
@@ -651,11 +661,14 @@ def formatrepr(self):
         if msg is None:
             fm = self.request._fixturemanager
             available = []
+            seen = set()
             parentid = self.request._pyfuncitem.parent.nodeid
             for name, fixturedefs in fm._arg2fixturedefs.items():
                 faclist = list(fm._matchfactories(fixturedefs, parentid))
-                if faclist:
+                name = strip_invocation_scope_suffix(name)
+                if faclist and name not in seen:
                     available.append(name)
+                    seen.add(name)
```

Comment:
I did it because list lookup is O(n) while set lookup is O(1). But now that you mentioned it I realize performance here is not important at all, so I think your suggestion is sound. Changed. :smile: